### PR TITLE
chore(elasticloadbalancingv2): add validation for `RedirectOptions.path`

### LINKED ISSUE #31201: elasticloadbalancingv2: add validation for path parameter of `ListenerAction.redirect()`
### Describe the feature

When setting the path property for `ListenerAction.redirect()`, CDK will validate whether it starts with a `/`.

### Use Case

If we set a string without a leading `/` for the path property, deployment will fail as shown below.

Given:

```ts
httpListener.addAction('RedirectAction', {
        conditions: [elbv2.ListenerCondition.pathPatterns(['/*'])],
        action: elbv2.ListenerAction.redirect({
          protocol: elbv2.ApplicationProtocol.HTTPS,
          path: 'example/path? ',
          permanent: true
        }),
      })
```

Result:

```sh
Failed resources:
MainStack-develop | 1:34:02 AM | CREATE_FAILED | AWS::ElasticLoadBalancingV2::ListenerRule | Alb/HttpListener/RedirectActionRule (AlbHttpListenerRedirectActionRule1D930694) Internal error reported from downstream service during operation 'The Path parameter must be a valid path, should start with a '/', and may contain up to one of each of these placeholders: '#{path}', '#{host}', '#{port}'. (Service: ElasticLoadBalancingV2, Status Code: 400, Request ID: fd4c7f01-9c97-44c1-a177-30ac04b2db26)'.
```

Related docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-redirectconfig.html#cfn-elasticloadbalancingv2-listenerrule-redirectconfig-path

### Proposed Solution

Add validation for path prop like below.

```ts
    if (options.path && !options.path.startsWith('/')) {
      throw new Error('Redirect path must start with a \'/\'');
    }
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.150.0

### Environment details (OS name and version, etc.)

irrelevant

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #31201 .

### Reason for this change

If we set a string without a leading `/` for the path property, deployment will fail as shown below.

Given:

```ts
httpListener.addAction('RedirectAction', {
        conditions: [elbv2.ListenerCondition.pathPatterns(['/*'])],
        action: elbv2.ListenerAction.redirect({
          protocol: elbv2.ApplicationProtocol.HTTPS,
          path: 'example/path? ',
          permanent: true
        }),
      })
```

Result:

```sh
Failed resources:
MainStack-develop | 1:34:02 AM | CREATE_FAILED | AWS::ElasticLoadBalancingV2::ListenerRule | Alb/HttpListener/RedirectActionRule (AlbHttpListenerRedirectActionRule1D930694) Internal error reported from downstream service during operation 'The Path parameter must be a valid path, should start with a '/', and may contain up to one of each of these placeholders: '#{path}', '#{host}', '#{port}'. (Service: ElasticLoadBalancingV2, Status Code: 400, Request ID: fd4c7f01-9c97-44c1-a177-30ac04b2db26)'.
```

Related docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-redirectconfig.html#cfn-elasticloadbalancingv2-listenerrule-redirectconfig-path

### Description of changes

Add validation for `path` prop like below.

```ts
    if (options.path && !options.path.startsWith('/')) {
      throw new Error('Redirect path must start with a \'/\'');
    }
```

### Description of how you validated changes

Add unit test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by badmintoncryer** [comment]
Exemption request:
Since it's only for adding validation, I believe there is no need to add integration test files.

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts [hunk 1]
```diff
@@ -115,6 +115,9 @@ export class ListenerAction implements IListenerAction {
     if ([options.host, options.path, options.port, options.protocol, options.query].findIndex(x => x !== undefined) === -1) {
       throw new Error('To prevent redirect loops, set at least one of \'protocol\', \'host\', \'port\', \'path\', or \'query\'.');
     }
+    if (options.path && !options.path.startsWith('/')) {
+      throw new Error('Redirect path must start with a \'/\'');
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts:119]: How about adding a feedback on an input like `..., got: ${options.path}`?

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts [hunk 2]
```diff
@@ -115,6 +115,9 @@ export class ListenerAction implements IListenerAction {
     if ([options.host, options.path, options.port, options.protocol, options.query].findIndex(x => x !== undefined) === -1) {
       throw new Error('To prevent redirect loops, set at least one of \'protocol\', \'host\', \'port\', \'path\', or \'query\'.');
     }
+    if (options.path && !options.path.startsWith('/')) {
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts:118]: Could you take the token case into account?

**Review by go-to-k — CHANGES_REQUESTED** [review]
Thanks for this PR. I made a few comments.

**Comment by badmintoncryer** [comment]
@go-to-k Thank you for your review!! I've addressed your comments.

**Review by go-to-k — APPROVED** [review]
LGTM.

**Review by GavinZZ — APPROVED** [review]

**Comment by badmintoncryer** [comment]
@mergify update

**Comment by badmintoncryer** [comment]
@GavinZZ Thank you for your approval. Could you please resolve CI failure?

**Comment by badmintoncryer** [comment]
@mergify update

**Pull request merged**